web/src/app/tokens/tokens-list/tokens-list.component.ts

```diff
@@ -34,18 +38,30 @@ export class TokensListComponent {
 
   // This function delete the project token
   delete(token: ProjectTokenModel): void {
-    this.projectTokenService
-      .delete(this.projectUid, token.uid)
-      .pipe(
-        take(1)
-      )
-      .subscribe(() => {
-        this.projectTokenService.findAll(this.projectUid)
+    let dialogConfig: MatDialogConfig = new MatDialogConfig();
+    dialogConfig = {
+      width: '500px',
+      data: {
+        title: 'Delete Token',
+        content: 'Are you sure you want to delete?',
+      },
+    };
+    this.dialogRef = this.dialog.open(DialogConfirmationComponent, dialogConfig);
+    this.dialogRef.afterClosed().subscribe((result: boolean) => {
+      if (result === true) {
+        this.projectTokenService
+          .delete(this.projectUid, token.uid)
           .pipe(
             take(1)
           )
-          .subscribe((data: ProjectTokenModel[]) => this.tokenList = data);
-      });
-  }
-
+          .subscribe(() => {
```

Comment:
Nested subscribe are not recommended. Its better to chain off the first pipe.

This seems different to the other dialogs that use a redirect after a delete? I think its better without a redirect if possible.